### ISSUE TITLE
SchemaMapping should apply Settings.defaultCatalog even if Settings.defaultSchema doesn't apply

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13034]
`SchemaMapping.map(Schema)` currently applies the `Settings.defaultCatalog` only if the `Settings.defaultSchema` also applies as can be seen in this logic:

```java
// The configured default schema is mapped to "null". This prevents
// it from being rendered to SQL
if ("".equals(result.getName())
    || result.getName().equals(mapping().getDefaultSchema())
        && (result.getCatalog() == null
            || "".equals(result.getCatalog().getName())
            || result.getCatalog().getName().equals(mapping().getDefaultCatalog())))
    result = null;
```

For example, with:

- `defaultCatalog = "C"`
- `defaultSchema = "S"`

This works:

```java
assertNull(mapping.map(schema(name("C", "S"))));
```

But this doesn't work:

```java
assertEquals(schema(name("X")), mapping.map(schema(name("C", "X"))));
```

In most cases it doesn't matter because if the resulting `Schema` is non-null, we render it, and then `SchemaMapping.map(Catalog)` applies, which gets it right. But if called directly, and in `AbstractRoutine` as well as `Function` (see https://github.com/jOOQ/jOOQ/issues/13033), we're currently not rendering the schema, but only the identifier.

**Comment by lukaseder** [comment]
This fix is implemented along with https://github.com/jOOQ/jOOQ/issues/13033, which won't be backported, so this fix will not be backported as well.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0
